lib/x86_cpu/x86_env.ml

```diff
@@ -197,6 +254,58 @@ module R64 = struct
   (* All YMM/XMM registers are available *)
   let ymms = ymms
 
+  (* mapping registers to arch variables *)
+  let of_reg reg =
+    let internal_reg = X86_asm_reg.decode reg in
+    match internal_reg with
+    | None -> None
+    | Some reg ->
+      (match reg with
+       | `AL | `AH | `AX | `EAX | `RAX -> Some rax
```

Comment:
I think that this code could be reused from the x86 representation.

Reply:
The `rax`, etc reference different objects. I can introduce sharing of course, but the function would have to be parameterized by the architecture. is that preferable?

Reply:
yep, I don't think that it would be to heavy to parametrize it with the modvars module. It could be hidden internally or made public, if preferred.